src/plugins/data/common/index_patterns/fields/field_list.ts

```diff
@@ -44,87 +45,76 @@ export type CreateIndexPatternFieldList = (
   onNotification?: OnNotification
 ) => IIndexPatternFieldList;
 
-export class FieldList extends Array<IndexPatternField> implements IIndexPatternFieldList {
-  private byName: FieldMap = new Map();
-  private groups: Map<IndexPatternField['type'], FieldMap> = new Map();
-  private indexPattern: IndexPattern;
-  private shortDotsEnable: boolean;
-  private onNotification: OnNotification;
-  private setByName = (field: IndexPatternField) => this.byName.set(field.name, field);
-  private setByGroup = (field: IndexPatternField) => {
-    if (typeof this.groups.get(field.type) === 'undefined') {
-      this.groups.set(field.type, new Map());
+export const fieldList = (
```

Comment:
nit: Would be nice to have a comment here explaining the need for wrapping the class in a provider.